management_interface/management_interface/test_views.py

```diff
@@ -0,0 +1,49 @@
+import json
+from http import HTTPStatus
+
+from django.test import TestCase
+from django.urls import reverse
+
+from .models import RegisteredManager
+
+
+class CareProviderLocationTests(TestCase):
+    def setUp(self) -> None:
+        self.manager = RegisteredManager.objects.create(
+            given_name="Jehosephat", family_name="McGibbons", cqc_registered_manager_id="My CQC RegsiteredManagerID"
+        )
+        self.location = self.manager.careproviderlocation_set.create(
+            name="My Location Name",
+            email="nosuchaddress@nhs.net",
+            ods_code="My Ods Code",
+            cqc_location_id="My CQC Location ID",
+        )
+        self.care_recipient = self.location.carerecipient_set.create(
+            subscription_id="42", provider_reference_id="foobar"
+        )
+        self.care_recipient.nhs_number = "password"
+        self.care_recipient.save()
+
```

Comment:
Think we can pull out something useful here:

```suggestion
    def assertFailure(self, response, status_code, code):
        self.assertEqual(response.status_code, status_code)
        self.assertEqual(response.json()["issue"][0]["code"], code)
```
The `response` object has a `json` method that lets us avoid the `import json` at the top, which is quite handy.

Reply:
Oh, also for the record I think it's fine for us not to be asserting on the `diagnostics` field here.  I don't think it would gain us very much.